### PR TITLE
chore(download): add optimize 3.11.5 to download page

### PR DESCRIPTION
**Pull request by misiekhardcore**

### CODE REVIEW

**Comment by misiekhardcore** [comment]
I dont think we have to update the `announcements.md` and I dont think the supported engine/ES versions have changed?

File: enterprise/content/download.md [hunk 1]
```diff
@@ -3620,13 +3620,13 @@ To install the demo distribution, please download one of the following archives:
   </thead>
   <tbody>
     <tr class="well">
-      <td>3.11.4</td>
+      <td>3.11.5</td>
       <td>7.18+, 7.19+, 7.20+</td>
       <td>8.8.0+</td>
       <td>12.12.2023</td>
```
Comment (OmranAbazid) [inline enterprise/content/download.md:3626]: We need to update the release as well. here and in the other places as well

Comment (misiekhardcore) [reply enterprise/content/download.md:3626]: what do you mean? I can't find any other references to `3.11.4` in this repo

File: enterprise/content/download.md [hunk 2]
```diff
@@ -3620,13 +3620,13 @@ To install the demo distribution, please download one of the following archives:
   </thead>
   <tbody>
     <tr class="well">
-      <td>3.11.4</td>
+      <td>3.11.5</td>
       <td>7.18+, 7.19+, 7.20+</td>
       <td>8.8.0+</td>
-      <td>12.12.2023</td>
+      <td>03.01.2024</td>
```
Comment (RomanJRW) [inline enterprise/content/download.md:3626]: shouldn't this be the 9th?

Comment (misiekhardcore) [reply enterprise/content/download.md:3626]: The release date is 9th of January?

Comment (RomanJRW) [reply enterprise/content/download.md:3626]: Yes, it's always the second Tuesday of the month. But we need to have it ready by the Wednesday before in order to allow downstream components to do their part

**Pull request merged**